src/app/dataset-view/additional-components/dataset-settings-component/tabs/webhooks/create-edit-subscription-modal/create-edit-subscription-modal.component.ts

```diff
@@ -0,0 +1,112 @@
+    DatasetBasicsFragment,
+    WebhookSubscriptionInput,
+    WebhookSubscriptionStatus,
+} from "src/app/api/kamu.graphql.interface";
+import { WebhooksService } from "src/app/services/webhooks.service";
+import { BaseComponent } from "src/app/common/components/base.component";
+import { takeUntilDestroyed } from "@angular/core/rxjs-interop";
+import {
+    CreateWebhookSubscriptionSuccess,
+    SubscribedEventType,
+    WebhookSubscriptionModalAction,
+} from "./create-edit-subscription-modal.model";
+import { MaybeNull, MaybeUndefined } from "src/app/interface/app.types";
+import { Clipboard } from "@angular/cdk/clipboard";
+import { changeCopyIcon } from "src/app/common/helpers/app.helpers";
+import { ErrorSets } from "src/app/common/directives/form-validation-errors.types";
+
+@Component({
+    selector: "app-create-edit-subscription-modal",
+    templateUrl: "./create-edit-subscription-modal.component.html",
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CreateEditSubscriptionModalComponent extends BaseComponent implements OnInit {
+    @Input({ required: true }) public datasetBasics: DatasetBasicsFragment;
+    @Input() public subscriptionData: MaybeNull<CreateWebhookSubscriptionSuccess>;
+
+    public dropdownList: SubscribedEventType[] = [];
+    public secret: MaybeUndefined<string>;
+    public readonly WebhookSubscriptionStatus: typeof WebhookSubscriptionStatus = WebhookSubscriptionStatus;
+    public readonly WebhookSubscriptionModalAction: typeof WebhookSubscriptionModalAction =
+        WebhookSubscriptionModalAction;
+    public readonly ErrorSets: typeof ErrorSets = ErrorSets;
+
+    public activeModal = inject(NgbActiveModal);
+    private fb = inject(FormBuilder);
+    private webhooksService = inject(WebhooksService);
+    private clipboard = inject(Clipboard);
+
+    public ngOnInit(): void {
+        this.webhooksService
+            .eventTypes()
+            .pipe(takeUntilDestroyed(this.destroyRef))
+            .subscribe((eventTypes: string[]) => {
+                this.dropdownList = eventTypes.map((eventType) => {
+                    return { name: this.eventTypesMapper(eventType), value: eventType };
+                });
+                if (this.subscriptionData) {
+                    this.createSubscriptionForm.patchValue({
+                        targetUrl: this.subscriptionData.input.targetUrl,
+                        eventTypes: this.subscriptionData.input.eventTypes,
+                        label: this.subscriptionData.input.label,
+                    });
+                    this.secret = this.subscriptionData.secret;
+                }
+            });
+    }
+
+    public get targetUrlControl(): AbstractControl {
+        return this.createSubscriptionForm.controls.targetUrl;
+    }
+
+    public get eventTypesControl(): AbstractControl {
+        return this.createSubscriptionForm.controls.eventTypes;
+    }
+
+    public createSubscriptionForm = this.fb.nonNullable.group({
```

Comment:
The name `createSubscriptionForm` is confusing, because we also edit here, not just create it.

Reply:
Renamed